src/utils/internalLinks.js

```diff
@@ -145,10 +145,10 @@ export const getNarrowFromNarrowLink = (
 
   if (
     // 'dm' is new in server-7.0; means the same as 'pm-with'
-    (hashSegments.length === 2 && (hashSegments[0] === 'pm-with' || hashSegments[0] === 'dm'))
-    || (hashSegments.length === 4
-      && (hashSegments[0] === 'pm-with' || hashSegments[0] === 'dm')
-      && hashSegments[2] === 'near')
```

Comment:
Hmm, I'd rather not have this regress in being specific about what it accepts, even though the existing code is already vague about this in the case of a stream-and-topic link. My previous comment was a bit unclear but what I meant was just fixing this to accept `/with/NNN` properly, not making it bug-compatible with the stream-and-topic case's willingness to accept `/arbitrary/nonsense` :slightly_smiling_face: 

Should be easy to do, just having it look for either string, the same way as in `getNearOperandFromLink` below.